src/shared/tpm2-util.c

```diff
@@ -6800,6 +6802,39 @@ int tpm2_pcrlock_search_file(const char *path, FILE **ret_file, char **ret_path)
         return 0;
 }
 
+int tpm2_pcrlock_policy_from_json(
+                JsonVariant *v,
+                Tpm2PCRLockPolicy *ret_policy) {
+
+        JsonDispatch policy_dispatch[] = {
+                { "pcrBank",    JSON_VARIANT_STRING,        json_dispatch_tpm2_algorithm, offsetof(Tpm2PCRLockPolicy, algorithm),       JSON_MANDATORY },
+                { "pcrValues",  JSON_VARIANT_ARRAY,         json_dispatch_variant,        offsetof(Tpm2PCRLockPolicy, prediction_json), JSON_MANDATORY },
+                { "nvIndex",    _JSON_VARIANT_TYPE_INVALID, json_dispatch_uint32,         offsetof(Tpm2PCRLockPolicy, nv_index),        JSON_MANDATORY },
```

Comment:
why is this TYPE_INVALID?

Reply:
Since 67a3028555c58cea152cd8b8e863a643eb147a97 we allow numbers to be specified as strings, to deal with the 53bit integer issue of json. because of that the type column in JsonDispatch tables should be specified as _JSONV_ARIANT_TYPE_INVALID (which means: don#t make an early type check), so thtat both integers and strings are passed to the integer parsers which can do the right thing then.

Reply:
Got it - I see it's a preexisting issue, but a comment would be helpful as it's not obvious

Reply:
i can certainly add a comment, but note that we generally want this to be done across the codebase, and we generally do already. sounds noisy adding such a comment to all such tables...

Reply:
Maybe a more descriptive alias for `_JSON_VARIANT_TYPE_INVALID` should be defined for this. `JSON_VARIANT_UNCHECKED_INTEGER` or something like that.

Reply:
That's a nice idea